androidApp/src/main/java/com/example/tasky/android/agenda/viewmodel/AgendaDetailsViewModel.kt

```diff
@@ -176,8 +176,26 @@ class AgendaDetailsViewModel(
     }
 
     private fun toggleIsGoing() {
+        val agendaItem = screenStateFlow.value.agendaItem
+
+        if (agendaItem !is Event) {
+            return
+        }
+
         viewModelScope.launch {
-            // TODO
+            val userId = SessionManager.getUserId() ?: ""
```

Comment:
Does the empty text here make sense? If there is no session then wouldn’t null represent this, by indicating the value does not exist?